### PR TITLE
Pull requests

### PR DESCRIPTION
**Pull request by rashmirathi1729**

### CODE REVIEW

File: lib/bitbucket_rest_api/repos/pull_request.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# encoding: utf-8
+# Mohd Irfanencoding: utf-8
```
Comment (dmgarland) [inline lib/bitbucket_rest_api/repos/pull_request.rb]: Was is das? Vim command gone wrong :)

Comment (rashmirathi1729) [inline lib/bitbucket_rest_api/repos/pull_request.rb]: Yep, fixed

File: lib/bitbucket_rest_api/repos/pull_request.rb [hunk 2]
```diff
@@ -38,5 +38,123 @@ def participants(user_name, repo_name, pull_request_id, params={})
...
+
+    def comment(user_name, repo_name, pull_request_id, comment_id, params={})
+      _update_user_repo_params(user_name, repo_name)
+      _validate_user_repo_params(user, repo) unless user? && repo?
+      normalize! params
+
+      response = request(:get, "/2.0/repositories/#{user}/#{repo.downcase}/pullrequests/#{pull_request_id}/comments/#{comment_id}", params)
+      return response unless block_given?
+    end
   end # Repos::Keys
```
Comment (dmgarland) [inline lib/bitbucket_rest_api/repos/pull_request.rb]: Not the DRYest class I've seen you guys write. The first 3 lines of each of these methods needs extracting into a private method

File: lib/bitbucket_rest_api/request.rb [hunk 3]
```diff
@@ -28,7 +28,7 @@ def delete_request(path, params={}, options={})
       request(:delete, path, params, options)
     end
 
-    def request(method, path, params, options)
+    def request(method, path, params, options={})
```
Comment (dmgarland) [inline lib/bitbucket_rest_api/request.rb]: Bit of whitespace around the `=` wouldn't go amiss

File: spec/bitbucket_rest_api/repos/pull_request_spec.rb [hunk 4]
```diff
@@ -0,0 +1,272 @@
+require 'spec_helper'
+
+describe BitBucket::Repos::PullRequest do
+  subject { described_class.new }
+  describe '#list' do
```
Comment (dmgarland) [inline spec/bitbucket_rest_api/repos/pull_request_spec.rb]: new line

**Comment by dmgarland** [comment]
Merged via master

**Pull request closed without merging**